client/my-sites/plans-features-main/hooks/use-plan-types-with-intent.ts

```diff
@@ -0,0 +1,105 @@
+import {
+	TYPE_BLOGGER,
+	TYPE_BUSINESS,
+	TYPE_ECOMMERCE,
+	TYPE_ENTERPRISE_GRID_WPCOM,
+	TYPE_FREE,
+	TYPE_PERSONAL,
+	TYPE_PREMIUM,
+	getPlan,
+	isBloggerPlan,
+} from '@automattic/calypso-products';
+
+export type Intent =
+	| 'blog-onboarding'
+	| 'newsletter'
+	| 'link-in-bio'
+	| 'new-hosted-site'
+	| 'new-hosted-site-hosting-flow'
+	| 'plugins'
+	| 'jetpack-app' // newly added
+	| 'import'
+	| 'default';
+
+interface Props {
+	intent: Intent;
+	selectedPlan: string;
+	sitePlanSlug: string;
+	hideEnterprisePlan: boolean;
+}
+
+interface PlanTypesWithIntent {
+	intent: Intent;
```

Comment:
I was confused by this hook since it accepts `intent` and returns `intent`.

Essentially it is casting unknown values of `intent` to `default`. I think this logic should be a part of `useIntentFromSiteMeta` and then `usePlanTypesWithIntent` can return just the `planTypes`.

Reply:
Definitely. No need for this to pass back the `intent` value. It's always passed in as either `default` or something else. Unless I overlooked something, there's no casting per se i.e. `return { intent: 'default', ... }` below may as well be `return { intent, ... }`. I think it just lingered this way from the hook doing more originally than it does now (it would derive the site intent at some point).